java/src/com.zeroc.ice/src/main/java/com/zeroc/Ice/ProtocolPluginFacade.java

```diff
@@ -5,25 +5,25 @@
 /**
  * @hidden Public because it's used by IceBT and the 'Ice/background' test.
  */
-public interface ProtocolPluginFacade {
-    //
-    // Get the Communicator instance with which this facade is
-    // associated.
-    //
-    Communicator getCommunicator();
+public final class ProtocolPluginFacade {
+    private final Instance _instance;
 
-    //
-    // Register an EndpointFactory.
-    //
-    void addEndpointFactory(EndpointFactory factory);
+    public ProtocolPluginFacade(Communicator communicator) {
+        _instance = communicator.getInstance();
+    }
 
-    //
-    // Get an EndpointFactory.
-    //
-    EndpointFactory getEndpointFactory(short type);
+    // Registers an EndpointFactory.
+    public void addEndpointFactory(EndpointFactory factory) {
+        _instance.endpointFactoryManager().add(factory);
+    }
 
-    //
-    // Look up a Java class by name.
-    //
-    Class<?> findClass(String className);
+    // Gets an EndpointFactory.
+    public EndpointFactory getEndpointFactory(short type) {
+        return _instance.endpointFactoryManager().get(type);
+    }
+
+    // Looks up a Java class by name.
+    public Class<?> findClass(String className) {
+        return _instance.findClass(className);
```

Comment:
Can we also get rid of this findClass method. It's just a wrapper around the instance one which is a wrapper around the static `Util.findClass` which is "public"